geoengine/layers.py

```diff
@@ -0,0 +1,687 @@
+        raise GeoEngineException(response.json())
+
+
+def _delete_layer_collection_from_collection(parent_id: LayerCollectionId,
+                                             collection_id: LayerCollectionId,
+                                             timeout: int = 60) -> None:
+    '''Delete a layer collection from a collection'''
+
+    session = get_session()
+
+    response = req.delete(
+        f'{session.server_url}/layerDb/collections/{parent_id}/collections/{collection_id}',
+        headers=session.admin_auth_header,
+        timeout=timeout,
+    )
+
+    if not response.ok:
+        raise GeoEngineException(response.json())
+
+
+def _delete_layer_collection(collection_id: LayerCollectionId,
+                             timeout: int = 60) -> None:
+    '''Delete a layer collection'''
+
+    session = get_session()
+
+    response = req.delete(
+        f'{session.server_url}/layerDb/collections/{collection_id}',
+        headers=session.admin_auth_header,
+        timeout=timeout,
+    )
+
+    if not response.ok:
+        raise GeoEngineException(response.json())
+
+
+def _add_layer_collection_to_collection(name: str,
+                                        description: str,
+                                        parent_collection_id: LayerCollectionId,
+                                        timeout: int = 60) -> LayerCollectionId:
+    '''Add a new layer collection'''
+
+    session = get_session()
+
+    response = req.post(
+        f'{session.server_url}/layerDb/collections/{parent_collection_id}/collections',
+        headers=session.admin_auth_header,
+        json={
+            "name": name,
+            "description": description,
+        },
+        timeout=timeout,
+    )
+
+    if not response.ok:
+        raise GeoEngineException(response.json())
+
+    return LayerCollectionId(UUID(response.json()['id']))
+
+
+def _add_existing_layer_collection_to_collection(collection_id: LayerCollectionId,
+                                                 parent_collection_id: LayerCollectionId,
+                                                 timeout: int = 60) -> None:
+    '''Add an existing layer collection to a collection'''
+
+    session = get_session()
+
+    response = req.post(
+        f'{session.server_url}/layerDb/collections/{parent_collection_id}/collections/{collection_id}',
+        headers=session.admin_auth_header,
+        timeout=timeout,
+    )
+
+    if not response.ok:
+        raise GeoEngineException(response.json())
+
+
+def _add_layer_to_collection(name: str,
+                             description: str,
+                             workflow: Dict[str, Any],  # TODO: improve type
+                             symbology: Optional[Dict[str, Any]],  # TODO: improve type
+                             collection_id: LayerCollectionId,
+                             timeout: int = 60) -> LayerId:
+    '''Add a new layer'''
+    # pylint: disable=too-many-arguments
+
+    session = get_session()
```

Comment:
has to be adjusted to new route

Reply:
I already did that, it is `f'{session.server_url}/layerDb/collections/{collection_id}/layers',`